### ISSUE TITLE
visit option is not getting passed to bbc-a11y

### ISSUE THREAD

**Issue by jameslawson** [bbc/a11y-tests-web#9]
## Summary

The [`visit` option](https://github.com/bbc/bbc-a11y/blob/5c8d01ae77833978ff4ab3b8b848459c2cdffae5/guides/using/using-bbc-a11y-in-your-project.md#advanced-usage-manually-setting-up-the-browser-window) in `options` does not work on in a11y-tests-web.
Assuming a minimal config where are _not_ using `signedInPaths`:

```javascript
module.exports = {
  options: {
    visit: function (frame) {
        /* code */
    },
    hide: [],
    skip: []
  },
  baseUrl: 'http://localhost:3000',
  paths: [ '/' ],
  signedInPaths: [ ]
};
```

## Expected Behaviour

The `visit` option works in bbc-a11y but does not work in a11-tests-web. It would be great to have this because we would like to inject a script to wait for the page to fully render.

The output sent to bbc-a11y would ideally be like this:
```
page( "http://localhost:3000/", { "hide":[],"skip":[], "visit": function() { ... } })
```

## Current Behaviour

The visit option is currently not getting passed to bbc-a11y. This is because a [`JSON.stringify`](https://github.com/bbc/a11y-tests-web/blob/74331af6f5d3f50c91e5edfcfc12ce9b959c4ca5/lib/bbcA11y.js#L53) is stripping out the function. The built-in behaviour of JSON.stringify is to [remove functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify):

> If undefined, a Function, or a Symbol is encountered during conversion it is either omitted (when it is found in an object) or censored to null (when it is found in an array). 

The output sent to bbc-a11y currently looks like:
```
page( "http://localhost:3000/", { "hide":[],"skip":[] })
```

## Possible Solution

There's probably a nicer way, but this, for example, seems to fix the problem:
```diff
   return `
     page(
       "${baseUrl}${path}",
       {
         ${visitOptions}
+        visit: ${options.visit.toString()},
         ${JSON.stringify(options).slice(1, -1)}
       }
     )
   `;
```

We make a change to what `pathToOutput` returns – we explicitly take the function body of `options.visit` and convert it to a string, outside of the `JSON.stringify`.

## Your Environment

* Version used: 3.1.0
* Operating System and version: macOS Sierra Version 10.12.6 (Google Chrome Version 66.0)
* Link to your project: *(testing a11y-tests-web locally for now)*

**Comment by andymsuk** [comment]
Hi @jameslawson.
Thanks for raising the issue and for all the details.
For some reason, I didn't get notified that you raised this issue so I've only just noticed it, but I will look into it ASAP. In the meantime, feel free to open a PR so fix this as you suggested and I can take a look!
Cheers,
Andy

**Comment by jameslawson** [comment]
Thanks @andymsuk !
